perception/alars/auv_detector/auv_detector/hook_estimator.py

```diff
@@ -0,0 +1,1020 @@
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+import cv2
+import numpy as np
+from cv_bridge import CvBridge
+from std_msgs.msg import Float32MultiArray, Header
+from geometry_msgs.msg import PoseStamped, AccelStamped, Vector3Stamped
+from sensor_msgs.msg import Imu
+from drone_msgs.msg import Topics as DroneTopics
+import auv_detector.params_detector_2 as P
+from collections import deque
+
+from tf2_ros import TransformListener, Buffer
+from geometry_msgs.msg import TransformStamped, PointStamped
+from tf2_ros import TransformException
+
+from scipy.spatial.transform import Rotation
+import matplotlib.pyplot as plt
+import matplotlib
+matplotlib.use('Agg')
+from collections import deque
+import os
+import csv
+from itertools import groupby
+from operator import itemgetter
+from filterpy.kalman import ExtendedKalmanFilter
+
+from scipy.spatial.transform import Rotation
+from datetime import datetime
+
+
+class hook_estimator(Node):
+    def __init__(self):
+        super().__init__('k_nearest_neighbors')
+
+        self.subscription_camera = self.create_subscription(
+            Image,
+            '/Quadrotor/core/fpcamera/image',
```

Comment:
use the Topics.msg system. Hardcoded topics are not allowed.
The first part `/Quadrotor` is the name of the robot, and your node should work for _any robot name_ so it will be namespaced by ros. None of your topics should start with `/` and none should have the robot's name in them.

For example, this topic parameter should be `DjiTopics.CAMERA_TOPIC` after importing `from dji_msgs.msg import Topics as DjiTopics`. Take this as an example for all things related to topics: https://github.com/smarc-project/smarc2/blob/humble/vehicles/hardware/dji/dji_captain/dji_captain/dji_captain.py